packages/cli/src/cmds/compare/ChangeReporter.ts

```diff
@@ -222,23 +258,91 @@ export default class ChangeReporter {
     assert(this.appMapMetadata);
     const failedAppMaps = new Set<AppMapName>();
     {
-      for (const appmap of await this.appmapData.appmaps(RevisionName.Head)) {
+      for (const appmap of await this.paths.appmaps(RevisionName.Head)) {
         const metadata = this.appMapMetadata[RevisionName.Head].get(appmap);
         if (!metadata) {
           console.log(prominentStyle(`Metadata for ${appmap} not found!`));
           continue;
         }
-        if (metadata.test_status === 'failed') {
-          this.referencedAppMaps.add(RevisionName.Head, appmap);
-          failedAppMaps.add(appmap);
-        }
+        if (metadata.test_status === 'failed') failedAppMaps.add(appmap);
       }
     }
     this.failedAppMaps = failedAppMaps;
   }
 }
 
-class ReportGenerator {
```

Comment:
These functions are new factored out of `ReportGenerator`, so that it's more clear what the data/functional dependencies of each function are.